### ISSUE TITLE
Integrate with hound?

### ISSUE THREAD

**Issue by carols10cents** [Manishearth/rust-clippy#838]
I promise to elaborate in a minute, just wanted to start this before I forget, please ping me and tell me to explain myself if I don't in the next 24 hrs ;)

https://github.com/houndci/hound

**Comment by Manishearth** [comment]
Also, https://codeclimate.com/

**Comment by llogiq** [comment]
Well, we have https://clippy.bashy.io so far, but I guess the more the merrier :smile:

What about the license? IANAL, so is MPL compatible with MIT (which seems to be what hound is using)?

**Comment by Manishearth** [comment]
I bet hound can use clippy as an executable without having an issue with the license. MPL is pretty permissive anyway.

**Comment by carols10cents** [comment]
> Well, we have https://clippy.bashy.io so far, but I guess the more the merrier :smile:

Correct me if I'm wrong, but it looks like the clippy service runs only on the specified branch of your project and reports the current results. Hound runs various linting tools on each pull request and comments on lines of the diff where it finds problems, so that the PR submitter can fix them *before* the code gets merged to the master branch.

I think codeclimate is more like the clippy service, in that it provides a score of the current state?

**Comment by carols10cents** [comment]
> Also, https://codeclimate.com/

Should integrating with codeclimate get its own issue?

**Comment by llogiq** [comment]
Feel free to open a new issue if you think integration is sufficiently different that it warrants seperate attention.

**Comment by Manishearth** [comment]
> Correct me if I'm wrong, but it looks like the clippy service runs only on the specified branch of your project and reports the current results

This is correct, we're not saying that this is an invalid issue, just noting similar services :smile: 

I'd love a per-PR check thing personally.

**Comment by Ryan1729** [comment]
If I'm not mistaken homu/bors currently addresses everything mentioned here. So this issue could be closed?

**Comment by Ryan1729** [comment]
Ah, yes. Bors currently comments on the issue if CI fails, but pointing to the problem in the PR diff and/or suggesting a new diff to fix the issue would be nicer experiences.

**Comment by brunoarueira** [comment]
There are an issue on houndci/hound#1344 to track the integration with clippy too, I'll give it a try on the next couple days.
